### PR TITLE
 feat(refactor): split runtime and static types

### PR DESCRIPTION
**Pull request by liubog2008**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
- split config types and runtime types
- refactor some directory structs
- change _context.yaml to context.yaml
```

### CODE REVIEW

**Comment by ddysher** [comment]
@zjj2wry to review

File: cleaner/cleaner.go [hunk 1]
```diff
@@ -12,9 +12,5 @@ type Cleaner interface {
 
 	// Clean will be called after all of the cases in the context are
 	// finished
-	Clean(variables map[string]jsonutil.Variable) error
-
-	// ForceClean will be called if function clean failed
-	// If ForceClean returns false, test case will fail
-	// ForceClean(variables map[string]jsonutil.Variable) bool
+	Clean(template *runtime.RoundTripTemplate, args map[string]string) error
```
Comment (zjj2wry) [inline cleaner/cleaner.go]: 每一个目录是一个 runtime.RoundTripTemplate？clean 用来清理某个目录下的变量？

File: framework/context.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+	ginkgo.By(originRoundTrip.Description)
+
+	respMatcher, err := roundtrip.MatchResponse(rt)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	if rt.Response.Async {
+		timeout := rt.Response.Timeout
+		interval := rt.Response.Interval
+
+		if timeout == time.Duration(0) {
```
Comment (zjj2wry) [inline framework/context.go]: timeout should be >interval?

File: framework/context.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+		}, timeout, interval).Should(respMatcher)
+	} else {
+		resp, err := gf.client.DoRequest(rt)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(resp).To(respMatcher)
+	}
+	vs := respMatcher.Variables()
+	for k, v := range vs {
+		_, ok := ctx.Variables[k]
+		gomega.Expect(ok && !overwrite).To(gomega.BeFalse(), "variable %v has been defined", k)
```
Comment (zjj2wry) [inline framework/context.go]: 后面的这个 description 真的是 format,args 格式吗？我看他们参数是一样的

File: framework/context.go (hunk 2 above)
Comment (liubog2008) [reply framework/context.go]: yes

File: framework/context.go (hunk 3 above)
Comment (liubog2008) [reply framework/context.go]: yes, I have seen source code...

File: cleaner/cleaner.go (hunk 1 above)
Comment (liubog2008) [reply cleaner/cleaner.go]: 1. template can also be inherited and overrided
2. `Cleaner` is used to clean context, e.g. clean database or call DELETE api

**Comment by ddysher** [comment]
@liubog2008 @zjj2wry what's the status?

**Comment by ddysher** [comment]
discussed offline with @liubog2008, this can be merged now

/lgtm

**Pull request merged**